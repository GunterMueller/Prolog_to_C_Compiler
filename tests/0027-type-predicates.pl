main :-
	var(X), display(var(X)), nl,
	\+var(1), display(\+var(1)), nl,
	\+var(a), display(\+var(a)), nl,
	\+var(x(y)), display(\+var(x(y))), nl,
	\+var(1.23), display(\+var(1.23)), nl,
	\+var([]), display(\+var([])), nl,
	\+nonvar(X), display(\+nonvar(X)), nl,
	nonvar(1), display(nonvar(1)), nl,
	nonvar(a), display(nonvar(a)), nl,
	nonvar(x(y)), display(nonvar(x(y))), nl,
	nonvar(1.23), display(nonvar(1.23)), nl,
	nonvar([]), display(nonvar([])), nl,
	\+number(X), display(\+number(X)), nl,
	number(1), display(number(1)), nl,
	\+number(a), display(\+number(a)), nl,
	\+number(x(y)), display(\+number(x(y))), nl,
	number(1.23), display(number(1.23)), nl,
	\+number([]), display(\+number([])), nl,
	\+compound(X), display(\+compound(X)), nl,
	\+compound(1), display(\+compound(1)), nl,
	\+compound(a), display(\+compound(a)), nl,
	compound(x(y)), display(compound(x(y))), nl,
	\+compound(1.23), display(\+compound(1.23)), nl,
	\+compound([]), display(\+compound([])), nl,
	\+atom(X), display(\+atom(X)), nl,
	\+atom(1), display(\+atom(1)), nl,
	atom(a), display(atom(a)), nl,
	\+atom(x(y)), display(\+atom(x(y))), nl,
	\+atom(1.23), display(\+atom(1.23)), nl,
	atom([]), display(atom([])), nl,
	\+atomic(X), display(\+atomic(X)), nl,
	atomic(1), display(atomic(1)), nl,
	atomic(a), display(atomic(a)), nl,
	\+atomic(x(y)), display(\+atomic(x(y))), nl,
	atomic(1.23), display(atomic(1.23)), nl,
	atomic([]), display(atomic([])), nl,
	\+integer(X), display(\+integer(X)), nl,
	integer(1), display(integer(1)), nl,
	\+integer(a), display(\+integer(a)), nl,
	\+integer(x(y)), display(\+integer(x(y))), nl,
	\+integer(1.23), display(\+integer(1.23)), nl,
	\+integer([]), display(\+integer([])), nl,
	\+float(X), display(\+float(X)), nl,
	\+float(1), display(\+float(1)), nl,
	\+float(a), display(\+float(a)), nl,
	\+float(x(y)), display(\+float(x(y))), nl,
	float(1.23), display(float(1.23)), nl,
	\+float([]), display(\+float([])), nl,
	seeing(S), is_stream(S), display(stream(s)), nl,
	\+is_stream(1), display(\+stream(1)), nl.
