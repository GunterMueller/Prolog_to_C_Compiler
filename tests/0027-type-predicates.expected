var(_0)
\+(var(1))
\+(var(a))
\+(var(x(y)))
\+(var(1.23))
\+(var([]))
\+(nonvar(_0))
nonvar(1)
nonvar(a)
nonvar(x(y))
nonvar(1.23)
nonvar([])
\+(number(_0))
number(1)
\+(number(a))
\+(number(x(y)))
number(1.23)
\+(number([]))
\+(compound(_0))
\+(compound(1))
\+(compound(a))
compound(x(y))
\+(compound(1.23))
\+(compound([]))
\+(atom(_0))
\+(atom(1))
atom(a)
\+(atom(x(y)))
\+(atom(1.23))
atom([])
\+(atomic(_0))
atomic(1)
atomic(a)
\+(atomic(x(y)))
atomic(1.23)
atomic([])
\+(integer(_0))
integer(1)
\+(integer(a))
\+(integer(x(y)))
\+(integer(1.23))
\+(integer([]))
\+(float(_0))
\+(float(1))
\+(float(a))
\+(float(x(y)))
float(1.23)
\+(float([]))
stream(s)
\+(stream(1))
